### ISSUE TITLE
Provide multiple paths containing pictures per user

### ISSUE THREAD

**Issue by totti4ever** [LibrePhotos/librephotos#123]
Hey there, this piece of software looks and feels great!

What I am looking for though is something where my wife and I can use the same system to view our pictures. From what I understand librephotos separates users completely, which is totally fine for me. Still, I would need to be able to add several places in the filesystem where pictures are scanned as we have a common folder with shared pictures and both a folder with own pictures.

Any chance?

keep up the great work!

**Comment by robertcontois** [comment]
There are multiple ways to do this. The easiest is to store multiple scan directories on the user record and then break it out later when it's needed, but that will create problems down the road. The proper way would be to create a new model to store directory locations associated with the user record. This would allow us to easily add features to individual directories in the future.

**Comment by derneuere** [comment]
I like the second option better. There will be a need for new features like ignoring certain filetypes or folders.

**Comment by robertcontois** [comment]
> I like the second option better. There will be a need for new features like ignoring certain filetypes or folders.

I agree. I like the concept of the PR for excluding particular directories/files, but that would be handled better somewhere like this rather than in the environment file.

**Comment by robertcontois** [comment]
Actually, this would help with performance issues to give the ability to disable features per directory too.

If I have 50,000 landscape photos, and I don't care about faces, I can mark them as not needing to run that scan on them. Or any of the other advanced features that might not be needed.

**Comment by totti4ever** [comment]
Sounds great! Any idea how to handle same directories being owned by more than one person? Is adding it twice and having it scanned etc twice the only option?

Though, for a quick workaround I simply mapped multiple folders to the data directory by editing the `docker-compose.yml`.

**Comment by derneuere** [comment]
You can share an album between two user. I think that would be the way to go. Adding it twice and having it scanned twice would not work I think, because it can't handle duplicates yet.

**Comment by totti4ever** [comment]
Even if its for two different users? How would it know about duplicates? Checksum? How is librephotos supposed to deal with such a situation when two users add the same picture independently (because a common friend sent it to both)?

**Comment by derneuere** [comment]
Currently, the primary key of the photo is a md5 hash of the photo. You can't have two identical primary keys in a database. We know that this is an issue, and we are talking about migrating it: https://github.com/LibrePhotos/librephotos/wiki/Open-Discussion:-Primary-key-and-migration-of-primary-key
